### ISSUE TITLE
Baseline comparisons incorrectly give FAIL result when baseline directory is missing

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#314]
First: Note that testing to demonstrate this issue was done with cime5.0.2, which I know isn't the latest. Apologies if it has been fixed since then.

When I run a test suite with baseline comparisons, the baseline comparison result is given as a BFAIL if the baseline directory exists but there are no files in it, but the result is given as a FAIL if the baseline directory does not exist at all. The latter should generate a BFAIL rather than a fail.

To be more specific: I compared against a baseline tag named 'baseline_test'. I then ran two tests with a compare tag of 'baseline_test':

SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel
SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel

I setup the 'baseline_test' directory with an empty directory corresponding to one of these tests:

```
[yslogin4:/glade/scratch/sacks/cesm_baselines/baseline_test]$ ls
SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel
```

Then I got the following TestStatus results:

```
PASS SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel INIT
PASS SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel CREATE_NEWCASE
PASS SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel XML
PASS SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel SETUP
NLFAIL SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel nlcomp
PASS SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel SHAREDLIB_BUILD
PASS SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel MODEL_BUILD
PASS SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel RUN
COMMENT: insuffiencient data for memleak test
COMMENT for SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel COMPCOMPAREbaseline history file does not exist : cism.h.nc : baseline compare baseline cism.h base with baseline_test
COMMENT for SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel COMPCOMPAREbaseline history file does not exist : cpl.hi.nc : baseline compare cpl.hi base with baseline_test
BFAIL SMS_D_Ly1.f09_g16_gl20.TGIS2.yellowstone_intel compare
COMMENT COMPCOMPAREbaseline history file does not exist : baseline compare summary
COMMENT TIME 49
```

```
PASS SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel INIT
PASS SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel CREATE_NEWCASE
PASS SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel XML
PASS SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel SETUP
NLFAIL SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel nlcomp
PASS SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel SHAREDLIB_BUILD
PASS SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel MODEL_BUILD
PASS SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel RUN
COMMENT: insuffiencient data for memleak test
FAIL SMS_D_Ly1.f09_g16_gl5.TG.yellowstone_intel compare

COMMENT TIME 178
```

The FAIL... compare line in the second TestStatus should be a BFAIL.

**Comment by jgfouca** [comment]
@billsacks : How about keeping it FAIL and just adding a comment that the reason for the failure was missing baselines?

**Comment by billsacks** [comment]
@jgfouca : I'd prefer keeping BFAIL or some other term. This is because there are often times when I run a test suite and know that I only have baselines for a subset of the tests; in these cases, it's helpful to be able to quickly filter out the BFAIL results so I'm just left with the true failures.

That said, I'd be happy with your suggestion as long as the comment appears on the FAIL line and is worded / formatted in a way that makes it easily grep'able - i.e., so I can still filter out those results with a 'grep -v' or something similar. I think the main requirement is that there be some unique string that appears in this situation and no other.

**Comment by jgfouca** [comment]
Yes, it will be on the same line in order to remain in the spirit of the other issue you opened.

**Comment by jgfouca** [comment]
Fixed by #350 , dunno why issue didn't auto-close.
